### DISCUSSION TITLE
react v4 server cache

### DISCUSSION THREAD

**Discussion by samik3k** [TanStack/query#5882] [Q&A]
Hello all. I have a couple questions.

Im using v4.24.4 with SSR and nextjs without app folder

1) So, i added getServerSideProps with `await queryClient.prefetchQuery(` and `dehydratedState: dehydrate(queryClient)`
At page like page/[pageId]. When im changing pageId, same component, just content different, server cache working?
```js
export async function getServerSideProps() {
  await queryClient.prefetchQuery(
    [ SOME_KEY ],
    getSomething(),
    {
      ...CACHE_SETTINGS,
    },
  )

  return {
    props: {
      dehydratedState: dehydrate(queryClient),
    },
  }
}
```
I mean, that when i will change router for different pageId, will it get data from cache?

2) We have some strange problems with server cache (i think so)
We using 1 docker file to turn on 3 different build of project and sometime we get strange data in react-query.
Its like it merged between build.
We will try to disable server cache for test. But its very strange

**Comment by TkDodo** [comment]
no, react-query is per se not a server cache. Every time you call `getServerSideProps`, you should create a new QueryClient, as shown in the docs:

```diff
export async function getServerSideProps() {
+  const queryClient = new QueryClient()
   await queryClient.prefetchQuery(...)
```

**Reply by samik3k** [reply]
Yes, i have that line

**Comment by samik3k** [comment]
For now i found one more problem.
So, i have some api, im getting array with 2 items.
But after react-query im getting like 14 items. Its like react-query getting some cache from memory.
I have cacheTime: 0 in useQuery options
